src/pragmas/gcc.zig

```diff
@@ -126,7 +127,16 @@ fn preprocessorHandler(_: *Pragma, pp: *Preprocessor, start_idx: TokenIndex) Pra
                     directive_tok.expansionSlice(),
                 );
             },
-            .diagnostic => return diagnosticHandler(pp, start_idx + 2),
+            .diagnostic => return diagnosticHandler(pp, start_idx + 2) catch |err| switch (err) {
+                error.UnknownPragma => {
+                    const tok = pp.tokens.get(start_idx + 2);
+                    return pp.comp.diag.add(.{
+                        .tag = .unknown_pragmas_directive,
+                        .loc = tok.loc,
+                    }, tok.expansionSlice());
+                },
+                else => |e| return e,
```

Comment:
Why is this here and not in `diagnosticHandler`?

Reply:
Because doing it inside ``diagnosticHandler`` prints the warning message twice. I think that's because the function is also used in ``parserHandler`` when pragma syntax error do not need to be checked.

Reply:
You could add a bool parameter to `diagnosticHandler` called `diagnose_unknown` (or a 2-element enum with `diagnose_unknown` & `ignore_unknown`) that is true when called from `preprocessorHandler` and false when called from `parserHandler`, which controls whether `unknown_pragmas_directive` gets issued